tools/src/main/java/com/nedap/archie/creation/OpenEhrRmInstanceGenerator.java

```diff
@@ -0,0 +1,375 @@
+                    break;
+                case 2: //percent
+                    result.put("denominator", 100);
+                    break;
+                case 3: //fraction
+                case 4: //integer fraction
+                    result.put("precision", 0);
+            }
+        }
+    }
+
+    private void FixPartyRelated(Map<String, Object> result) {
+        Object name = result.get("name");
+        if(name == null) {
+            result.put("name", "John Doe");
+        }
+    }
+
+    private void fixDvMultimedia(Map<String, Object> result) {
+        Object data = result.get("data");
+        if(data == null) {
+            result.put("data", "NDIK");
+        }
+    }
+
+    private void fixElement(Map<String, Object> result) {
+        Object value = result.get("value");
+        Object nullFlavour = result.get("null_flavour");
+        if(value == null && nullFlavour == null) {
+            Map<String, Object> dvText = new LinkedHashMap<>();
+            dvText.put(typePropertyName, "DV_TEXT");
+            dvText.put("value", "string");
+
+            result.put("value", dvText);
+        }
+        if(value != null && nullFlavour != null) {
+            result.remove("null_flavour");//can't be set at the same time
+        }
+    }
+
+    private void fixHistory(Map<String, Object> result) {
+        Object events = result.get("events");
+        if(events == null || (events instanceof List && ((List) events).isEmpty())) {
+            List<Map> newEvents = new ArrayList<>();
+            newEvents.add(this.generator.constructExampleType("EVENT"));
+            result.put("events", newEvents);
+        }
+    }
+
+    private void fixCodePhrase(Map<String, Object> result, CObject cObject) {
+        try {
+            String codeString = (String) result.get("code_string");//TODO: check terminology code to be local?
+
+            if(AOMUtils.isValueCode(codeString)) {
+                //check for OpenEHR term mapping and use that if available, so we get correct
+                //rm objects
+
+                //TODO: quite some more term ids, such as IANA characters sets, etc. HOW?
+                URI openehr = generator.getArchetype().getTerminology(cObject).getTermBinding("openehr", codeString);
+                if(openehr != null && openehr.getPath() != null) {
+                    int i = openehr.getPath().lastIndexOf('/');
+                    if (i > 0) {
+                        codeString = openehr.getPath().substring(i+1);
+                        Map<String, Object> terminologyId = (Map<String, Object>) result.get("terminology_id");
+                        if(terminologyId != null) {
+                            terminologyId.put("value", "openehr");
+                        }
+                        result.put("code_string", codeString);
+                        //TODO: add a mapping to the at code in the data.
+                    }
+                }
+            }
+        } catch (Exception e) {
```

Comment:
I'm not sure what you're trying to catch here, but would it be possible to catch a more specific exception here? The tests don't seem to hit this, so maybe it even can be removed?

Reply:
This is old code that I just moved from one class to a better location. But indeed, easy to fix, will do.